app/models/data_export.rb

```diff
@@ -57,15 +58,18 @@ def to_csv!
     save!
   end
 
-  private
-
-  def set_name
+  def set_default_name
```

Comment:
Name had been set in a before_save. The goal of moving it to this public method is for it to be set and available to show in the `new` form.

Reply:
Yeah, in my original design the form included a preset dropdown or something for type, so parts of the name weren't resolvable when the form was rendered. I think I'd prefer a `default_name` method here that's public, show it as the placeholder for the input (instead of prefilling the input) and then have `set_name` remain in the `before_save` as `self.name ||= default_name`